extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -278,4 +326,162 @@ public void transform(TransformationContext ctx) {
+            LaunchModeBuildItem launchMode,
+            LiveReloadBuildItem liveReload,
+            HttpRootPathBuildItem httpRootPath,
+            CurateOutcomeBuildItem curateOutcomeBuildItem) throws Exception {
+
+        if (!health.enableUi) {
+            return;
+        }
+        if ("/".equals(health.rootPathUi)) {
+            throw new ConfigurationError(
+                    "quarkus.smallrye-health.root-path-ui was set to \"/\", this is not allowed as it blocks the application from serving anything else.");
+        }
+
+        String healthPath = httpRootPath.adjustPath(health.rootPath);
+
+        if (launchMode.getLaunchMode().isDevOrTest()) {
+            CachedHealthUI cached = liveReload.getContextObject(CachedHealthUI.class);
+            boolean extractionNeeded = cached == null;
+
+            if (cached != null && !cached.cachedHealthPath.equals(healthPath)) {
+                try {
+                    FileUtil.deleteDirectory(Paths.get(cached.cachedDirectory));
+                } catch (IOException e) {
+                    LOG.error("Failed to clean Health UI temp directory on restart", e);
+                }
+                extractionNeeded = true;
+            }
+            if (extractionNeeded) {
+                if (cached == null) {
+                    cached = new CachedHealthUI();
+                    liveReload.setContextObject(CachedHealthUI.class, cached);
+                    Runtime.getRuntime().addShutdownHook(new Thread(cached, "Health UI Shutdown Hook"));
+                }
+                try {
+                    AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+                    Path tempDir = Files.createTempDirectory(TEMP_DIR_PREFIX).toRealPath();
+                    extractHealthUi(artifact, tempDir);
+                    cached.cachedDirectory = tempDir.toAbsolutePath().toString();
+                    cached.cachedHealthPath = healthPath;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            Handler<RoutingContext> handler = recorder.uiHandler(cached.cachedDirectory,
+                    httpRootPath.adjustPath(health.rootPathUi));
+            routeProducer.produce(new RouteBuildItem(health.rootPathUi, handler));
+            routeProducer.produce(new RouteBuildItem(health.rootPathUi + "/*", handler));
+            notFoundPageDisplayableEndpointProducer
+                    .produce(new NotFoundPageDisplayableEndpointBuildItem(health.rootPathUi + "/"));
+        } else if (health.alwaysIncludeUi) {
+            AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+            //we are including in a production artifact
+            //just stick the files in the generated output
+            //we could do this for dev mode as well but then we need to extract them every time
+            for (Path p : artifact.getPaths()) {
```

Comment:
can `extractHealthUi` be reused here?

Reply:
Not really no, they do different things.